docs/transient-crr-source.md

```diff
@@ -0,0 +1,177 @@
+# Transient CRR Source
+
+## Description
+
+The goal of this enhancement is to save the cost of reading data from
+a public cloud one or more times for CRR purpose, when both the
+primary and CRR targets are on public clouds.
+
+## Requirements
+
+It must be possible to configure object locations and CRR targets to
+store objects on multiple public clouds (aka. external targets)
+without a permanent local storage target, while being able to read
+locally for CRR purpose, avoiding the cost of reading from public
+clouds for CRR.
+
+## Design
+
+### Design overview
+
+Normal CRR process reads from the source defined by the location
+constraint, writes to one or more destinations asynchronously and
+updates statuses to *COMPLETED* on the source when done.
+
+We are going to keep this core behavior, but introduce the notion of
+*transient source* as a temporary primary location for a new object,
+until the CRR completes the processing of this object.
+
+The following will then happen when an object is put to a location
+which is configured as being a transient source:
+
+* the object will first be put to the location which is configured as
+  being a transient source, then the put status is returned as usual,
+
+* CRR processors will transfer the object asynchronously to all CRR
+  targets set through replication API (also as usual),
+
+* finally, once all CRR targets have been processed, the object
+  location will be altered to be the preferred (default) CRR target
+  location, and the transient data will be deleted asynchronously by a
+  garbage collection process.
+
```

Comment:
Maybe add some explanation about what happens in the case that one of the backends fails. Do we still want to clean up the transient data? Perhaps in that case we can accept any future read from the updated location?